Q1. What are the three measures of central tendency?

Mean, Median, Mode are the three measures of central tendency.

Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

Mean - The mean is the sum of the value of each observation in a dataset divided by the number of observations. This is also known as the arithmetic average. 

Median - The median is the middle value in distribution when the values are arranged in ascending or descending order.

Mode - The mode is the most commonly occurring value in a distribution.

Mean is used to know the middle value of the dataset. But when outliers are present in the database, then we median is used to find the central tendency. Mode is used to find the maximum frequency of the dataset.

Q3. Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [4]:
import numpy as np
from scipy import stats
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
print("Mean: ",np.mean(data))
print("Median: ",np.median(data))
print("Mode: ",stats.mode(data))

Mean:  177.01875
Median:  177.0
Mode:  ModeResult(mode=array([177.]), count=array([3]))


/tmp/ipykernel_70/2491601974.py:6: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  print("Mode: ",stats.mode(data))


Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [9]:
import numpy as np
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]
print("Standard Deviation: ",np.std(data))
print("Variance: ",np.var(data))
print("Range: ",max(data)-min(data))

Standard Deviation:  1.7885814036548633
Variance:  3.199023437500001
Range:  7.5


Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.

Range - The range is the easiest dispersion of data or measure of variability. The range can measure by subtracting the lowest value from the massive Number. The wide range indicates high variability, and the small range specifies low variability in the distribution.

Variance - Variance measures how far each number in the dataset from the mean. To compute variance first, calculate the mean and squared deviations from a mean.

Standard deviation - It is a squared root of the variance to get original values. Low standard deviation indicates data points close to mean.

Example - Let assume the list of height is [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]. Standard Deviation:  1.7885814036548633, Variance:  3.199023437500001, Range:  7.5

From the range of the dataset, we came to know the difference between highest and lowest value of the dataset.
From variance of the dataset, we came to know how far each number in the dataset from the mean.
From standard deviation, we came to know the skewness of the data.

Q6. What is a Venn diagram?

A Venn diagram illustrates the relationships between two or more data sets. Venn diagrams are especially useful for highlighting similarities and differences and are commonly used to compare and contrast the characteristics of different data sets.

Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A ⋂ B
(ii) A ⋃ B

(i) A ⋂ B - (2,6)
(ii) A ⋃ B - (0,2,3,4,5,6,7,8,10)

Q8. What do you understand about skewness in data?

Skewness is a measurement of the distortion of symmetrical distribution or asymmetry in a data set. Skewness is demonstrated on a bell curve when data points are not distributed symmetrically to the left and right sides of the median on a bell curve.

Q9. If a data is right skewed then what will be the position of median with respect to mean?

Mean is greater than equal to median for right skewed data. 

Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?

Covariance shows you how the two variables differ, whereas correlation shows you how the two variables are related.

Covariance tells us the direction of the relationship between two variables, while correlation provides an indication as to how strong the relationship between the two variables is, in addition to the direction of correlated variables.

Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.

Formula for calculating the sample mean is -> x̄ = ( Σ xi ) / n . Here, x̄ represents the sample mean, xi refers all X sample values and n stands for the number of sample terms in the data set.

Example: Dataset=[3,4,5,6,7]
         Sample Mean = (3+4+5+6+7)/5 = 25/5 = 5

Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Mean, Median, Mode are the three measures of central tendency.
Mean – Mode = 3 (Mean – Median)

Q13. How is covariance different from correlation?

Covariance tells us the direction of the relationship between two variables, while correlation provides an indication as to how strong the relationship between the two variables is, in addition to the direction of correlated variables."

Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Outliers are numbers in a data set that are vastly larger or smaller than the other values in the set. Mean, median and mode are measures of central tendency. Mean is the only measure of central tendency that is always affected by an outlier.

Example - Let's assume [2,4,5,6,1,100] is the dataset.
100 is the outlier of the dataset. Mean of the dataset is - 19.67. If we remove the outlier from the dataset, then mean will be 3. So, we can see that Mean is majorly affected by an outlier.